addon/modifiers/did-intersect.js

```diff
@@ -38,15 +68,58 @@ export default class DidIntersectModifier extends Modifier {
       return;
     }
 
-    let { onEnter, onExit, options } = this.args.named;
+    let { onEnter, onExit, maxEnter, maxExit, options } = this.args.named;
 
     assert('onEnter or/and onExit is required', onEnter || onExit);
 
-    if (onEnter) {
-      this.observerManager.addEnterCallback(this.element, onEnter);
+    // We should technically accept 0, in case of programmatic changes
+    if (maxEnter || maxEnter === 0) {
+      assert('maxEnter must be an integer', Number.isInteger(maxEnter));
+      this._maxEnterIntersections = maxEnter;
     }
-    if (onExit) {
-      this.observerManager.addExitCallback(this.element, onExit);
+
+    // We should accept 0 here too.
+    if (maxExit || maxExit === 0) {
+      assert('maxExit must be an integer', Number.isInteger(maxExit));
+      this._maxExitIntersections = maxExit;
+    }
+
+    if (onEnter && !this._isExceedingMaxEnters) {
+      this.observerManager.addEnterCallback(this.element, (...args) => {
+        if (this.isDestroying) {
+          return;
+        }
+
+        this._numOfEnters++;
+
+        if (this._isExceedingMaxEnters && this._isExceedingMaxExits) {
```

Comment:
Won't this be false if `maxEnters` has been exceeded but there is no `maxExits` callback? Same question for line 114 below

Reply:
I assume the intended behavior we want because `onExit` can fire unlimited times, if not provided an `maxExit`.
We just want to stop the `enter callback` but not completely unobserve the element?

Unless we don't want this to be the case.

Reply:
err I worded my question poorly. What if there is no `onExit` provided? We should handle that case. If there is no `onExit` at all and `maxEnters` is exceeded, we should unobserve (and vice versa).

Reply:
oh gotcha